File: 2.week/8-mars_exploration/go/main.go
package main

import (
    "bufio"
    "fmt"
    "io"
    "os"
    "strings"
)

/*
 * Complete the 'marsExploration' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts STRING s as parameter.
 */

func marsExploration(s string) int32 {
    // Write your code here
    
    strings.ToUpper(s)
    
    var changedCharacter int32
    
    for index, _ := range s {
        if ((index % 3 == 0 || index % 3 ==2) && s[index] != 0x53 ) {
            changedCharacter += 1
        } 
        if index % 3 == 1 && s[index] != 0x4F {
            changedCharacter += 1
        }
    } 
    
    return changedCharacter
}


func main() {
    reader := bufio.NewReaderSize(os.Stdin, 16 * 1024 * 1024)

    stdout, err := os.Create(os.Getenv("OUTPUT_PATH"))
    checkError(err)

    defer stdout.Close()

    writer := bufio.NewWriterSize(stdout, 16 * 1024 * 1024)

    s := readLine(reader)

    result := marsExploration(s)

    fmt.Fprintf(writer, "%d\n", result)

    writer.Flush()
}

func readLine(reader *bufio.Reader) string {
    str, _, err := reader.ReadLine()
    if err == io.EOF {
        return ""
    }

    return strings.TrimRight(string(str), "\r\n")
}

func checkError(err error) {
    if err != nil {
        panic(err)
    }
}

File: 2.week/8-mars_exploration/python/main.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'marsExploration' function below.
#
# The function is expected to return an INTEGER.
# The function accepts STRING s as parameter.
#

def marsExploration(s):
    # Write your code here
    s.upper()
    changedCharacter = 0
    for item in range(len(s)):
        if((item % 3 ==0 or item % 3 ==2) and s[item] != "S" ):
            changedCharacter +=1
        if(item % 3 ==1 and s[item] != "O" ):
            changedCharacter +=1

    print(changedCharacter)
    return changedCharacter

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    s = input()

    result = marsExploration(s)

    fptr.write(str(result) + '\n')

    fptr.close()

